### ISSUE TITLE
Non-model plastome assembly, no reads mapped bowtie2

### ISSUE THREAD

**Issue by mcsimenc** [mrmckain/Fast-Plast#13]
Hi! I want to use Fast-Plast to assemble cp genomes for a handful of non-model dicots. I have installed Fast-Plast but when I run it, the bowtie2 mapping step completes without mapping any reads. Can you help me figure out what is going on?

Here is the *Progess.log file contents:

```
Fri Jun  9 11:32:38 2017	Starting Fast-Plast v.1.2.6.
				Assemblying plastome with 0 single end libraries and 1 paired end libraries.
Fri Jun  9 11:32:38 2017	Determining best kmer sizes.
				K-mer sizes for SPAdes set at 55,87,121.
Fri Jun  9 11:32:39 2017	Starting read trimming with Trimmomatic.
				Using /home/derstudent/software/Trimmomatic-0.36/trimmomatic-0.36.jar.
Fri Jun  9 11:35:46 2017	Starting read mapping with bowtie2.
				Using bowtie2.
				Samples for All are not in current GenBank plastomes. Using one representative from each order to make bowtie2 indices.
Fri Jun  9 11:36:10 2017	Starting initial assembly with SPAdes.
				Using spades.py.
		******************ERROR: No mapped reads files were identified to run SPAdes.  Please check 2_Bowtie_Mapping.******************
```

Thank you!!

**Comment by mrmckain** [comment]
Can you check a couple of things?

1) Did Trimmomatic work? Are there reads in the 1_Trimmed_Reads directory? If not, rerun with the full path to your raw reads files and see if this takes care of the problem.

2) Are there bowtie2 indices in the 2_Bowtie_Mapping directory?  If not, let me know and I will check to make sure there isn't a bug in the code.

**Comment by mcsimenc** [comment]
Thanks for getting back 

1. Trimmomatic ran. There are *P1.fq, *P2.fq & *U.fq files in 1_Trimmed_reads/

2. No bowtie2 indices. There are 2 FASTA files in 2_Bowtie_Mapping/, one called All and one called All.fsa

**Comment by mrmckain** [comment]
Can you send the output of the error log?

**Comment by mcsimenc** [comment]
```
$ cat Agma.4051_results_error.log
TrimmomaticPE: Started with arguments:
 -threads 40 /home/joshd/data/santalales/nickrent_wgs/wgs_reads/Acanthosyris_asipapote_Nickrent4051b.fastq.gz /home/joshd/data/santalales/nickrent_wgs/wgs_reads/Acanthosyris_asipapote_Nickrent4051.fastq.gz Acas.4051_0.trimmed_P1.fq Acas.4051_0.trimmed_U1.fq Acas.4051_0.trimmed_P2.fq Acas.4051_0.trimmed_U2.fq ILLUMINACLIP:/home/joshd/software/Fast-Plast_with_afin/bin/adapters/NEB-PE.fa:1:30:10 SLIDINGWINDOW:10:20 MINLEN:40
Using Long Clipping Sequence: 'ACACTCTTTCCCTACACGACGCTCTTCCGATC'
Using Long Clipping Sequence: 'GATCGGAAGAGCACACGTCTGAACTCCAGTC'
ILLUMINACLIP: Using 0 prefix pairs, 2 forward/reverse sequences, 0 forward only sequences, 0 reverse only sequences
Quality encoding detected as phred33
Input Read Pairs: 4625323 Both Surviving: 4392854 (94.97%) Forward Only Surviving: 93347 (2.02%) Reverse Only Surviving: 87850 (1.90%) Dropped: 51272 (1.11%)
TrimmomaticPE: Completed successfully
No mapped reads files were identified to run SPAdes.  Please check 2_Bowtie_Mapping.
```

**Comment by mrmckain** [comment]
OK.  One more, can you send the output of *results_out.log?

**Comment by mcsimenc** [comment]
It's empty.

```
$ cat Agma.4970_results_out.log
$ ls -lh
total 10K
drwxr-xr-x 5 derstudent derlab  112 Jun  9 11:36 Agma.4970
-rw-r--r-- 1 derstudent derlab    0 Jun  9 11:32 Agma.4970.err
-rw-r--r-- 1 derstudent derlab  794 Jun  9 11:36 Agma.4970_Fast-Plast_Progress.log
-rw-r--r-- 1 derstudent derlab    0 Jun  9 11:32 Agma.4970.out
-rw-r--r-- 1 derstudent derlab 1.1K Jun  9 11:36 Agma.4970_results_error.log
-rw-r--r-- 1 derstudent derlab    0 Jun  9 11:32 Agma.4970_results_out.log
-rwxr-x--- 1 derstudent derlab 2.8K Jun  9 11:32 call_fastplast.qsub
```

**Comment by mrmckain** [comment]
I just pushed a bug fix for the "All" option with the bowtie2 indices.  Reinstall Fast-Plast and give it a try.  I just tested it and it worked for me.  

Also verify that bowtie2 is properly installed.

Best,
Michael

**Comment by mcsimenc** [comment]
Thanks Michael, I reinstalled Fast-Plast but am getting the same result. The species I'm working with are all in Santalales.

edit: bowtie2 is installed ok. Is there a bowtie2 command I should try outside of Fast-Plast?

**Comment by mrmckain** [comment]
I just checked and the code is working for on my install.  Here are some things for you to check.  If These are not the case, we will try something else.

1) Have you verified that bowtie2 is installed and working?  
2) Did you provided the full path to your raw reads?
3) Are there trimmed read files properly filled?
4) Since you are using paired end data, did you provide both R1 and R2, and not R1 or R2 twice by accident?

**Comment by mcsimenc** [comment]
Hmm ok I discovered that on lines 25-30 in my fast-plast.pl, where the paths to dependencies are defined, all dependencies except trimmomatic were define without full paths, like:

my $BOWTIE2="bowtie2"

I re-ran INSTALL.pl and provided the full paths (before, I had used PATH when asked for the paths), but fast-plast.pl was not changed. So I manually changed my fast-plast.pl so the full paths were in there.

I think I misunderstood what the installer did with when I used the PATH option to tell it where dependencies are–the dependencies were in the path at the time of installation but not when I actually ran fast-plast.pl. It's running now and has gotten past the bowtie2 step! Thanks!!
